src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java

```diff
@@ -125,4 +127,20 @@ public Session endSession(Long teamId) {
         endedSession.getStartDateTime(), endedSession.getEndDateTime());
     return endedSession;
   }
+
+  private boolean isSessionExpired(Session session) {
+    return session.getStartDateTime().plusHours(24).isBefore(LocalDateTime.now());
+  }
+
+  @Scheduled(fixedRate = 60000) // runs every minute
+  public void endExpiredSessions() {
+    log.debug("Checking for expired sessions...");
+    List<Session> sessions = sessionRepository.findAll();
+    for (Session session : sessions) {
+      if (isSessionExpired(session)
+          && session.getEndDateTime() == null) { // Check if the session is active
+        endSession(session.getTeam().getTeamId());
+      }
+    }
```

Comment:
too expensive. rather write a new repo method to get all sessions which have an empty end date field

Reply:
made a new repo method:
findByEndDateTimeIsNull()
@5v3nn